### PR TITLE
docs(contribute): adds table of contents and info about dependencies.

### PR DESCRIPTION
**Pull request by pranshuchittora**
- [x] Adds Table of Contents.
- [x] Adds a section about dependencies. 

**What kind of change does this PR introduce?**
Docs

**Motivation**
- It was getting harder to reach the desired section in the doc.
- CI failing while modifying dependency (not committing the `package-lock.json`)

Hence everything sorted  :tada:

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -5,6 +5,36 @@ appreciated and welcomed. If you're planning a new feature or changing
 the API, please create an issue first. This way we can ensure that your precious
 work is not in vain.
 
+Table of Contents
+
+- [Issues](#issues)
+- [Your first Contribution](#your-first-contribution)
+- [Setup](#setup)
+    - [Using npm (setup)](#using-npm-setup)
```
Comment (misterdev) [inline .github/CONTRIBUTING.md]: I would remove "(setup)" & "(testing)", I mean changing
- [Using npm (setup)](#using-npm-setup)
to
- [Using npm](#using-npm-setup)

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -5,6 +5,36 @@ appreciated and welcomed. If you're planning a new feature or changing
...
+- [Testing](#testing)
+    - [Using npm (testing)](#using-npm-testing)
+    - [Using yarn (testing)](#using-yarn-testing)
+- [Editor Config](#editor-config)
+- [Dependencies](#dependencies)
+- [Branching Model](#branching-model)
+- [Naming a branch](#naming-a-branch)
+    - [Features](#features)
+    - [Fixes](#fixes)
+- [Testing](#testing-1)
```
Comment (misterdev) [inline .github/CONTRIBUTING.md]: There's another entry named "Testing" in the table of contents

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -5,6 +5,36 @@ appreciated and welcomed. If you're planning a new feature or changing
...
+- [Pull Requests](#pull-requests)
+- [Submitting a good Pull Request](#submitting-a-good-pull-request)
+- [Commit message format](#commit-message-format)
+    - [Commit Message Format](#commit-message-format)
+- [Migrating with the CLI](#migrate-with-the-cli)
```
Comment (misterdev) [inline .github/CONTRIBUTING.md]: IMO we can keep "Migrating with the CLI" and remove the sub-entries from the TOC.
Also, this link doesn't work

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -5,6 +5,36 @@ appreciated and welcomed. If you're planning a new feature or changing
...
+- [Dependencies](#dependencies)
+- [Branching Model](#branching-model)
+- [Naming a branch](#naming-a-branch)
+    - [Features](#features)
+    - [Fixes](#fixes)
+- [Testing](#testing-1)
+- [Pull Requests](#pull-requests)
+- [Submitting a good Pull Request](#submitting-a-good-pull-request)
+- [Commit message format](#commit-message-format)
+    - [Commit Message Format](#commit-message-format)
```
Comment (misterdev) [inline .github/CONTRIBUTING.md]: Duplicated "Commit Message Format"

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -5,6 +5,36 @@ appreciated and welcomed. If you're planning a new feature or changing
...
+- [Migrating with the CLI](#migrate-with-the-cli)
+    - [How it's being done](#how-its-being-done)
+        - [Structure of a transform](#structure-of-a-transform)
+    - [TODO](#todo)
```
Comment (misterdev) [inline .github/CONTRIBUTING.md]: I would change TODO into something like "Further work"

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -5,6 +5,36 @@ appreciated and welcomed. If you're planning a new feature or changing
...
 
+Table of Contents
+
+
+
+- [Issues](#issues)
+- [Your first Contribution](#your-first-contribution)
+- [Setup](#setup)
+	- [Using npm](#using-npm)
+	- [Using yarn](#using-yarn)
```
Comment (ematipico) [inline .github/CONTRIBUTING.md]: The indentation is not correct. Could you please check with prettier? Or an online linter? https://dlaa.me/markdownlint/

Comment (pranshuchittora) [reply .github/CONTRIBUTING.md]: Done, pls join the discussion on #850

**Pull request merged**